main/src/ca/uwaterloo/flix/language/phase/TypeSimplifier.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2025 Jonathan Lindegaard Starup
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ca.uwaterloo.flix.language.phase
+
+import ca.uwaterloo.flix.language.ast.{Kind, Type}
+import ca.uwaterloo.flix.language.phase.typer.TypeConstraint
+import ca.uwaterloo.flix.language.phase.unification.EffUnification3
+import ca.uwaterloo.flix.util.InternalCompilerException
+
+/** This type simplification is purely focused on user readability. */
+object TypeSimplifier {
+
+  /**
+    * Simplifies types and effects - intended for user readability.
+    *
+    * This function should never crash and works for any type, no matter how ill-kinded.
```

Comment:
Something about returning an equivalent type (assuming that is true).

And also something about "how" it is simplified.

Reply:
Is it for effects? records what?

Reply:
I hope that we agree that formatting a type should not crash. The current implementation just doesn't live up to the interface

Reply:
But that seems to suggest you are introducing behavior that can crash the compiler?

Or is that already the case (?)

Reply:
The behavior is completely unchanged, just moved functions. If you call formatType on a broken type it can crash yea